lib/hooks/use-search.ts

```diff
@@ -1,27 +1,91 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
 
-export function useSearch<T = unknown[]>(items: T[], fieldNames: string[]) {
-  const [filteredItems, setFilteredItems] = useState<T[]>(items);
-  const handleSubmitKeywords = (keywords: string) => {
-    const lowerKeywords = keywords.toLowerCase();
-    setFilteredItems(
-      items.filter((item) => {
-        const filterBy = (fieldName: keyof T) => {
-          const selectedField = item[fieldName];
-
-          if (typeof selectedField === "string") {
-            return selectedField.toLowerCase().includes(lowerKeywords);
-          } else {
-            return false;
-          }
+type AggregationSetting = {
+  field: string;
+  title: string;
+};
+
+export function useSearch<T = unknown[]>(
+  items: T[],
+  fieldNames: string[],
+  aggregationSettings?: AggregationSetting[],
+  sortSettings?: {},
+  defaultSort?: string,
+) {
+  const configuration: any = {
+    searchableFields: fieldNames,
+    sortings: {
+      default: {
+        field: "id",
+        order: "asc",
+      },
+      ...sortSettings,
+    },
+  };
+  if (aggregationSettings?.length) {
+    configuration.aggregations = aggregationSettings.reduce(
+      (aggregations: any, cur) => {
+        aggregations[cur.field] = {
+          title: cur.title,
+          size: 10,
+          sort: "term",
+          order: "asc",
         };
+        return aggregations;
+      },
+      {},
+    );
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const itemsjs = require("itemsjs")(items, configuration);
```

Comment:
Can we import this library dynamically using the Next.js' [dynamic import](https://nextjs.org/docs/advanced-features/dynamic-import) feature here?

Reply:
`next/dynamic` itself is for Component based, so I can't use it for the library. Maybe it can be used when rendering `components/search-form.tsx` but it won't affect the search library importing.
Still can't figure out the right way to do it.

Reply:
Okay, I think we can defer lazy-loading the library for now, since it doesn't affect performance significantly.

However, why are you using `require` instead of the `import` statement? If you're using the `import` statement, you wouldn't need to disable the ESLint rule, would you?
What's stopping you from using the `import` statement?

Reply:
The library doesn't have type description so I couldn't use `import`, just added .d.ts file and changed `require` to `import`